experiments/AtmosLES/convective_bl_model.jl

```diff
@@ -0,0 +1,322 @@
+const param_set = EarthParameterSet()
+
+import ClimateMachine.Atmos: atmos_source!, flux_second_order!
+using ClimateMachine.Atmos: altitude, recover_thermo_state
+
+"""
+  ConvectiveBL Geostrophic Forcing (Source)
+"""
+struct ConvectiveBLGeostrophic{FT} <: Source
+    "Coriolis parameter [s⁻¹]"
+    f_coriolis::FT
+    "Eastward geostrophic velocity `[m/s]` (Base)"
+    u_geostrophic::FT
+    "Eastward geostrophic velocity `[m/s]` (Slope)"
+    u_slope::FT
+    "Northward geostrophic velocity `[m/s]`"
+    v_geostrophic::FT
+end
+function atmos_source!(
+    s::ConvectiveBLGeostrophic,
+    atmos::AtmosModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+
+    f_coriolis = s.f_coriolis
+    u_geostrophic = s.u_geostrophic
+    u_slope = s.u_slope
+    v_geostrophic = s.v_geostrophic
+
+    z = altitude(atmos, aux)
+    # Note z dependence of eastward geostrophic velocity
+    u_geo = SVector(u_geostrophic + u_slope * z, v_geostrophic, 0)
+    ẑ = vertical_unit_vector(atmos, aux)
+    fkvector = f_coriolis * ẑ
+    # Accumulate sources
+    source.ρu -= fkvector × (state.ρu .- state.ρ * u_geo)
+    return nothing
+end
+
+"""
+  ConvectiveBL Sponge (Source)
+"""
+struct ConvectiveBLSponge{FT} <: Source
+    "Maximum domain altitude (m)"
+    z_max::FT
+    "Altitude at with sponge starts (m)"
+    z_sponge::FT
+    "Sponge Strength 0 ⩽ α_max ⩽ 1"
+    α_max::FT
+    "Sponge exponent"
+    γ::FT
+    "Eastward geostrophic velocity `[m/s]` (Base)"
+    u_geostrophic::FT
+    "Eastward geostrophic velocity `[m/s]` (Slope)"
+    u_slope::FT
+    "Northward geostrophic velocity `[m/s]`"
+    v_geostrophic::FT
+end
+function atmos_source!(
+    s::ConvectiveBLSponge,
+    atmos::AtmosModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+
+    z_max = s.z_max
+    z_sponge = s.z_sponge
+    α_max = s.α_max
+    γ = s.γ
+    u_geostrophic = s.u_geostrophic
+    u_slope = s.u_slope
+    v_geostrophic = s.v_geostrophic
+
+    z = altitude(atmos, aux)
+    u_geo = SVector(u_geostrophic + u_slope * z, v_geostrophic, 0)
+    ẑ = vertical_unit_vector(atmos, aux)
+    # Accumulate sources
+    if z_sponge <= z
+        r = (z - z_sponge) / (z_max - z_sponge)
+        β_sponge = α_max * sinpi(r / 2)^s.γ
+        source.ρu -= β_sponge * (state.ρu .- state.ρ * u_geo)
+    end
+    return nothing
+end
+
+"""
+  Initial Condition for ConvectiveBoundaryLayer LES
+"""
+function init_convective_bl!(problem, bl, state, aux, localgeo, t)
+    (x, y, z) = localgeo.coord
+
+    # Problem floating point precision
+    FT = eltype(state)
+    R_gas::FT = R_d(bl.param_set)
+    c_p::FT = cp_d(bl.param_set)
+    c_v::FT = cv_d(bl.param_set)
+    p0::FT = MSLP(bl.param_set)
+    _grav::FT = grav(bl.param_set)
+    γ::FT = c_p / c_v
+    # Initialise speeds [u = Eastward, v = Northward, w = Vertical]
+    u::FT = 8
```

Comment:
This is also in the former file, but I think it is a typo. If you look at the forcing below, the geostrophic velocity is 4 m/s, not 8. Could you check with the paper @yairchn ?

Reply:
that is correct - good catch - I think the "8" stay there from using the Bomex file and changing